test/e2e/specs/all-features-confirmation/record/presentation.spec.js

```diff
@@ -163,4 +170,92 @@ describe('View existing record,', function() {
         });
     });
 
+    describe("For side panel table of contents in Record App", function() {
+
+        beforeAll(function() {
+            var url = browser.params.url + "/record/#" + browser.params.catalogId + "/" + testParams.sidePanelTest.schemaName + ":" + testParams.sidePanelTest.tableName +  "/id=" + testParams.sidePanelTest.id;
+            browser.get(url);
+            recSidePan = chaisePage.recordPage.getElementById('recordSidePan-5');
+            fiddlerBtn = element(by.className('sidePanFiddler')).element(by.tagName('i')),
+            chaisePage.waitForElement(fiddlerBtn);
+        });
+        it('Table of contents should be collapsed by default', function(){
+            var recordSidePan = chaisePage.recordPage.getElementById('record-side-pan');
+            recordSidePan.isDisplayed().then(function (bool) {
+                expect(bool).toBeFalsy();
+            });
+        });
+
+        it('Table of contents should match all related table', function(){
+            var  allPan = element.all(by.xpath("//div[contains(@id,'recordSidePan')]"));
+            chaisePage.waitForElement(fiddlerBtn).then(function(){
+              return fiddlerBtn.click();
+            }).then(function(){
+              expect(allPan.count()).toBe(testParams.sidePanelTest.tocCount, "Table of contents count did not match!");
+            }).catch( function(err) {
+               console.log(err);
+           });
+        });
+
+        it('On click of Related table name in TOC, page should move to the contents and open the table details', function(){
+            var rtTableHeading = chaisePage.recordPage.getRelatedTableHeading(testParams.sidePanelTest.tableToShow),
+                lastRow = element(by.id('rt-heading-Categories_5')).all(by.tagName('tr')).get(5);
+
+            recSidePan.getAttribute('class').then(function(className) {
+                return recSidePan.click();
+            }).then (function (){
+              // related table should be visible
+                rtTableHeading.isDisplayed().then(function (bool) {
+                  expect(bool).toBeTruthy();
+                });
+                chaisePage.waitForElement(lastRow);
```

Comment:
This `waitForElement` may not work properly if the inner data (the table rows) load after the panel opens. How this was done in other parts of our application was by waiting for a certain number of related table entries to be present before continuing.
```
browser.wait(function () {
  return chaisePage.recordPage.getREaltedTableRows(displayname).count().then(function (ct) {
    return (ct == numberOfRecords);
  });
});
```